lldbinit.py

```diff
@@ -9,16 +9,173 @@
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::cfg::Ident")
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::cfg::Send")
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::cfg::CFG")
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::cfg::VariableUseSite")
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::cfg::LocalRef")
+
+def register_summary(ci, funcName, typeName):
+    result = lldb.SBCommandReturnObject()
+    ci.HandleCommand("type summary add -F %s.%s %s" % (__name__, funcName, typeName), result)
+    return result
+
+def init_source_map(debugger, result):
+    # Bazel builds all binaries from a username-specific cache location
+    # for lldb to work properly we need to map that path back to our main source
+    project_root = os.path.dirname(os.path.realpath(__file__))
+    bazel_workspace = os.path.realpath(os.path.join(project_root, "bazel-sorbet"))
+
+    if not os.path.isdir(bazel_workspace):
+        result.Print("You need to compile Sorbet first before loading this file")
+    else:
+        result.Print("Mapping %s to current working directory" % bazel_workspace)
+        source_map_command = 'settings set -- target.source-map "%s" "%s"' % (bazel_workspace, project_root)
+        ci = debugger.GetCommandInterpreter()
+        res = lldb.SBCommandReturnObject()
+        ci.HandleCommand(source_map_command, res)
+
+def walk_frames_from(frame):
+    while frame is not None:
+        yield frame
+        frame = frame.get_parent_frame()
+
+def get_variable_in_frames(startFrame, typeNames):
+    if not isinstance(typeNames, list):
+        typeNames = [typeNames]
+    for (index, frame) in enumerate(walk_frames_from(startFrame)):
+        all_vars = frame.get_all_variables()
+        result = next((v for v in all_vars if any(typeName in v.GetTypeName() for typeName in typeNames)), None)
+        if result is None:
+            continue
+
+        resultType = result.GetType()
+        if result.TypeIsPointerType():
+            result = result.Dereference()
+        # Parameters passed by reference are a bit special because they can't be dereferenced like a pointer would
+        # be, but since they are really just pointers in disguise we can manually retrieve the underlying address
+        # they represent and reconstruct a value of the right type from it
+        if resultType.IsReferenceType():
+            error = lldb.SBError()
+            ref_data = result.GetData()
+            ref_addr = ref_data.GetAddress(error, 0)
+            if error.Success():
+                result = result.CreateValueFromAddress("__deref" + result.get_expr_path(), ref_addr, resultType.GetDereferencedType())
+        # If the variable was found in a parent frame we have to persist it globally to be able to access it
+        if index != 0:
+            result = result.Persist()
+        return result
+    return None
+
+def format_argument(valobj):
```

Comment:
Let's give this a more descriptive name than `format_argument`; `maybe_format_for_dereferencing`?  I'm not sure what color the bikeshed should be painted here.

Reply:
Hum, not sure either, I would want something more compact and generic than just say it's about dereferencing but I don't have a strong opinion.